connmgr/banmanager.go

```diff
@@ -125,10 +137,9 @@ func (bm *BanManager) BanPeer(p *peer.Peer) {
 	}
 
 	bm.mtx.Lock()
-	bmp, ok := bm.peers[p.Addr()]
-	if !ok {
+	bmp := bm.lookupPeer(p)
+	if bmp == nil {
```

Comment:
Since nothing else is using the lock, you can just unlock it before this if statement instead of the two places below.